src/api/spec/models/obs_factory/distribution_spec.rb

```diff
@@ -243,5 +243,37 @@ def strategy_for(name)
     it { expect(distribution.rings_project_name).to eq('openSUSE:Factory:Rings') }
   end
 
-  # TODO: OpenQA => openqa_filter and openqa_jobs_for
+  describe '#openqa_filter' do
+    let(:staging_project_a) { create(:project, name: 'openSUSE:Factory:Staging:A') }
+    let(:staging_project) { ObsFactory::StagingProject.new(project: staging_project_a, distribution: distribution) }
+
+    it { expect(distribution.openqa_filter(staging_project)).to eq('match=Staging:A') }
+  end
+
+  describe '#openqa_jobs_for' do
+    before do
+      allow(ObsFactory::OpenqaJob).to receive(:find_all_by)
+    end
+
+    shared_examples 'calls find_all_by' do |version, expected_build|
+      it "performs find_all_by with #{version} version" do
+        allow(distribution).to receive("#{version}_version").and_return(expected_build)
+        distribution.openqa_jobs_for(version)
+        expect(ObsFactory::OpenqaJob).to have_received(:find_all_by).with({ distri: 'opensuse', version: 'Tumbleweed', build: expected_build, group: 'openSUSE Tumbleweed' },
+                                                                          exclude_modules: true)
+      end
+    end
+
+    include_examples 'calls find_all_by', 'totest', '20180701'
+    include_examples 'calls find_all_by', 'published', '20180702'
+    include_examples 'calls find_all_by', 'source', '20180703'
+  end
+
+  describe '#totest_version' do
+    before do
+      allow(distribution.strategy).to receive(:totest_version).and_return('20180702')
+    end
+
+    it { expect(distribution.totest_version).to eq('20180702') }
```

Comment:
Do we need to test this? It's just delegating the method call to the strategy :thinking:

Reply:
You are right :thinking:  There was another similar test for '#published_version'. Should I remove it as well?